src/components/TextTruncated/index.tsx

```diff
@@ -0,0 +1,127 @@
+import { Link, Theme, Tooltip, Typography, TypographyProps } from '@mui/material';
+import { useState } from 'react';
+import { makeStyles } from '@mui/styles';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  arrow: {
+    color: `${theme.palette.TwClrBg} !important`,
+    '&:before': {
+      border: `1px solid ${theme.palette.TwClrBrdrTertiary}`
+    },
+  },
+  tooltip: {
+    backgroundColor: `${theme.palette.TwClrBg} !important`,
+    color: `${theme.palette.TwClrTxt} !important`,
+    border: `1px solid ${theme.palette.TwClrBrdrTertiary} !important`,
+    borderRadius: '8px !important',
+    boxShadow: `0px 3px 3px ${theme.palette.TwClrBaseGray100}`,
+    padding: '8px',
+  },
+}));
+
+export interface Props {
+  stringList: string[];
+  maxChars: number;
+  moreSeparator?: string;
+  moreText?: string;
+  textStyle?: TypographyProps['sx'];
+  showAllStyle?: TypographyProps['sx'];
+}
+
+/**
+ * Represents the result of truncating the string list
+ * text: the text to display before the '...# more' link
+ * numberMore: the number to display in the '...# more' link
+ *   has two special values
+ *   0: the full text can be displayed; do not show '...# more'
+ *   -1: the list is of length one but the element cannot be fully
+ *       displayed. show '...more' without a '#'
+ */
+interface TextDisplay {
+  text: string;
+  numberMore: number;
+}
+
+/**
+ * Compute TextDisplay for a list of strings
+ * @param strings list of strings to process
+ * @param maxLength max displayable length
+ */
+function computeFromStringList(
+  strings: string[],
+  maxLength: number,
+): TextDisplay {
+  if (strings[0].length > maxLength) {
+    // case in which the first string is too long to fit by itself
+    return {
+      text: strings[0].substring(0, maxLength),
+      numberMore: strings.length === 1 ? -1 : strings.length - 1
+    };
+  }
+
+  // concatenate strings with ', ' and stop before exceeding max
+  let stringsRemaining = strings.length - 1;
+  let result = strings[0];
+  for (let i = 1; i < strings.length; i++) {
+    const tempResult = [result, strings[i]].join(', ');
+    if (tempResult.length > maxLength) {
+      break;
+    }
+    stringsRemaining--;
+    result = tempResult;
+  }
+
+  return {
+    text: result,
+    numberMore: stringsRemaining,
+  };
+}
+
+export default function TextTruncated(props: Props): JSX.Element {
+  const {
+    stringList,
+    maxChars,
```

Comment:
Ideally the table cell renderer will call this component with a list of strings (maybe a ref to the table cell ?).
Component figures out how many characters can be rendered within the width.
Either that or we would need a helper component that can translate available width into maxChars etc here.
Bit of a catch-22 here since table cell width is determined by content displayed.. and here we want to display content by table cell width. A once popular technique is to render hidden elements outside the viewport to do some calculation. Perhaps @constanzauanini might have some suggestions.

Reply:
What I saw in some libraries is that the component accepts the number of lines to show (https://www.npmjs.com/package/react-truncate) and also the width. We could set fixed width for these cases when we need to truncate? I saw in this library that is has know issues when resizing 😕

Reply:
A suggestion, let me know your thoughts.

For initial version:
1) keep component as-is
2) have client pass in a width (as per Constanza's suggestion)
3) use a helper within web-components to map width to maxChars

Once we get this working, we can see if we still need to eliminate 2)..

Reply:
okay, that seems like a reasonable approach. Thank you both for your feedback and suggestions!